### PR TITLE
Mongo

### PR DESCRIPTION
**Pull request by garrypolley**
Adding in the ability to use mongo as a backend. 

Will need to update this to work with sql as well.  However, that is not in our current scope.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
 Whiskerboard is a status board for websites, services, and APIs, like Amazon's
 [AWS status page](http://status.aws.amazon.com/).
 
 It is heavily based on [Stashboard](http://www.stashboard.org/). Unlike
 Stashboard, it uses vanilla Django, so you aren't stuck using Google App Engine.
 
+## Mongo or Sql?
+This app supports using Mongo or SQL, howerver, the data models are quite different.  I recommend looking at each models.py
+file and taking a look to see which is best for you.  The reason for the difference is largely based on need at the time
+of creation, and the storage differences between sql and mongo.
```
Comment (RobertRyterski) [inline README.md]: I made issue #3 to bring the SQL and Mongo models to the same point. Might wanna change this to reflect that.

File: example/mongo_settings.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+AUTHENTICATION_BACKENDS = (
+    'mongoengine.django.auth.MongoEngineBackend',
+)
+
+SESSION_ENGINE = 'mongoengine.django.sessions'
+
+db_name = "cloud_status"
```
Comment (RobertRyterski) [inline example/mongo_settings.py]: Why not 'whiskerboard'?

File: example/settings.py [hunk 3]
```diff
@@ -22,8 +22,8 @@
     'django.contrib.sites',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    'django.contrib.admin',
```
Comment (RobertRyterski) [inline example/settings.py:26]: Does Django blow up with this and Mongo? Can the SQL version not have this anymore?

File: example/settings.py [hunk 4]
```diff
@@ -22,8 +22,8 @@
...
     'whiskerboard',
+    'mongonaut',
```
Comment (RobertRyterski) [inline example/settings.py:28]: Maybe we can mirror the sql_models and mongo_models setup, and only include this in mongo_settings. Then django.contrib.admin can be in sql_settings.

File: example/urls.py [hunk 5]
```diff
@@ -1,12 +1,10 @@
 from django.conf.urls import patterns, include, url
-from django.contrib import admin
+#from django.contrib import admin
 
-admin.autodiscover()
+#admin.autodiscover()
 
 urlpatterns = patterns('',
+    url(r'^mongonaut/', include('mongonaut.urls')),
```
Comment (RobertRyterski) [inline example/urls.py:7]: We'll have to do the dual setup here too.

File: whiskerboard/fixtures/initial_data.json [hunk 6]
```diff
@@ -1,35 +0,0 @@
```
Comment (RobertRyterski) [inline whiskerboard/fixtures/initial_data.json:-1]: We'll have to figure how to handle the initial data with #3.

File: whiskerboard/mongo_models.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+from django.template.defaultfilters import slugify
+from django.utils.translation import ugettext as _
+from mongoengine.document import Document
+from mongoengine.document import EmbeddedDocument
+from mongoengine.fields import DateTimeField
+from mongoengine.fields import EmbeddedDocumentField
+from mongoengine.fields import ListField
+from mongoengine.fields import StringField
+
+STATUS_CODES = {'ok': _('Ok'),
```
Comment (RobertRyterski) [inline whiskerboard/mongo_models.py]: `STATUS_CODES` is already defined in the base models.py.

File: README.md (hunk 1 above)
Comment (garrypolley) [inline README.md]: Not really.  The issues is created and viewable.  Most services will still work right now.

File: example/mongo_settings.py (hunk 2 above)
Comment (garrypolley) [inline example/mongo_settings.py]: you are correct.

File: example/settings.py (hunk 3 above)
Comment (garrypolley) [inline example/settings.py:26]: I just moved it out.  It should be there.

File: example/urls.py (hunk 5 above)
Comment (garrypolley) [inline example/urls.py:7]: example app.  I don't really care about making it not use it.

File: whiskerboard/fixtures/initial_data.json (hunk 6 above)
Comment (garrypolley) [inline whiskerboard/fixtures/initial_data.json:-1]: shouldn't require initial data.

File: whiskerboard/mongoadmin.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
```
Comment (RobertRyterski) [inline whiskerboard/mongoadmin.py:0]: Why not follow suit, and do mongo_admin.py?

File: whiskerboard/templates/whiskerboard/mongo_index.html [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    {% for service in services %}
+
+          <table id="service-list">
+            <thead>
+              <tr>
+                <th>{% trans 'Service' %}</th>
+                <th class="today">{% trans 'Current' %}</th>
+              
+                {% for day in past %}
+                  <th class="date">{{ day|date:"N j" }}</th>
```
Comment (RobertRyterski) [inline whiskerboard/templates/whiskerboard/mongo_index.html:19]: The date should get internationalized.

File: whiskerboard/mongoadmin.py (hunk 8 above)
Comment (garrypolley) [inline whiskerboard/mongoadmin.py:0]: mongonaut looks for mongoadmin.py

File: whiskerboard/templates/whiskerboard/mongo_index.html (hunk 9 above)
Comment (garrypolley) [inline whiskerboard/templates/whiskerboard/mongo_index.html:19]: this is the same in index.html.  Just copied this over. For now, I don't really care about this, until the use case for international dates are given.  But it's not that hard.

**Pull request merged**